src/entity/entity.js

```diff
@@ -276,14 +276,27 @@
         draw : function (renderer) {
             // draw the sprite if defined
             if (this.renderable) {
-                // translate the renderable position (relative to the entity)
-                // and keeps it in the entity defined bounds
-                var x = ~~(0.5 + this.pos.x + this.body.pos.x + (
-                    this.anchorPoint.x * (this.body.width - this.renderable.width)
-                ));
-                var y = ~~(0.5 + this.pos.y + this.body.pos.y + (
-                    this.anchorPoint.y * (this.body.height - this.renderable.height)
-                ));
+                var x = 0, y = 0;
+                if (this.renderable.hasTextureAnchorPoint) {
+                    // in this case, the entity's anchor point is in relation to the body
+                    // draw the renderable's anchorPoint at the Entity's anchor point
+                    x = ~~(0.5 + this.pos.x + this.body.pos.x + (
+                        (this.anchorPoint.x * this.body.width) - (this.renderable.anchorPoint.x * this.renderable.width)
+                    ));
+                    y = ~~(0.5 + this.pos.y + this.body.pos.y + (
+                        (this.anchorPoint.y * this.body.height) - (this.renderable.anchorPoint.y * this.renderable.height)
+                    ));
+                }
+                else {
+                    // translate the renderable position (relative to the entity)
+                    // and keeps it in the entity defined bounds
+                    x = ~~(0.5 + this.pos.x + this.body.pos.x + (
+                        this.anchorPoint.x * (this.body.width - this.renderable.width)
+                    ));
+                    y = ~~(0.5 + this.pos.y + this.body.pos.y + (
+                        this.anchorPoint.y * (this.body.height - this.renderable.height)
+                    ));
+                }
```

Comment:
If the anchorPoint translation moves to `me.Sprite`, it would be difficult for `me.Entity` to keep its current positioning behavior, where it completely ignores the anchorPoint and tries to position the edges of the sprite based on body width/height. 

I wasn't sure if it was OK to toss out the old positioning behavior. Some people may find it useful. Though, I think if we have both it makes the engine a little more confusing to use rather than committing to the new method.